starfish/imagestack/imagestack.py

```diff
@@ -290,6 +291,37 @@ def xarray(self) -> xr.DataArray:
         """Retrieves the image data as an xarray.DataArray"""
         return self._data
 
+    def _convert_to_indexers_dict(self, pos_kwargs):
+        return_dict = {ind.value: slice(None, None) for ind in Indices}
+        for key, value in pos_kwargs.items():
+            if isinstance(value, tuple):
+                value = slice(value[0], value[1])
+            return_dict[key] = value
+        return return_dict
+
+    def sel(self, indexers: Mapping[Indices, Union[int, tuple]]=None):
+        indexers = self._convert_to_indexers_dict(indexers)
+        indexed_data = self.index_keep_dimensions(self.xarray, indexers)
+        stack = self.from_numpy_array(indexed_data.data)
+        # set coords on new stack
+        stack._coordinates = physical_coordiante_calculator.calc_new_physical_coords(
+            self, indexers)
+        return stack
+
+    @staticmethod
+    def index_keep_dimensions(data: xr.DataArray, indexers) -> xr.DataArray:
```

Comment:
proposal: this method can stand alone outside of `ImageStack`.  it's a static method anyway.

Reply:
Is there a specific place you were thinking? Or should I add a new file to the util directory

Reply:
starfish/imagestack/??? 

I'm truly awful at naming things, so I won't even try. :)

Reply:
haha hmmm indexing_utils?